shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -196,6 +196,26 @@ public void send(@NonNull String channel, @Nullable ByteBuffer message, @Nullabl
   public void setMessageHandler(@NonNull String channel, @Nullable BinaryMessenger.BinaryMessageHandler handler) {
     messenger.setMessageHandler(channel, handler);
   }
+
+  /**
+   * Returns the number of pending channel callback replies.
+   *
+   * <p>When sending messages to the Flutter application using {@link BinaryMessenger#send(String,
+   * ByteBuffer, io.flutter.plugin.common.BinaryMessenger.BinaryReply)}, devlopers can optional
+   * specify a reply callback if they expect a reply from the Flutter application.
+   *
+   * <p>This method tracks all the pending callbacks that are waiting for response, and is supposed
+   * to be called from the main thread (as other methods). Calling from a different thread could
+   * possibly capture an indeterministic internal state, and is strongly discouraged.
```

Comment:
We can just say don't do it. No reason to even leave the door open to calling on a different thread, I don't think.

Reply:
Done.